### PR TITLE
CEM-2184 Referral Card Controller SDK

### PR DESCRIPTION
**Pull request by rmc3408**

### CODE REVIEW

File: src/app/controllers/ReferralCardController.ts [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+/**
+ * Controller related to Referal Card
+ */
+export class ReferralCardController {
+  app: App;
+  constructor(app: App) {
+    this.app = app;
+    // ADD BINDINGS BELOW
+    this.isValid = this.isValid.bind(this); /// Include this mutation ???????
```
Comment (ralph-dev) [inline src/app/controllers/ReferralCardController.ts:39]: Yes include it

File: src/app/controllers/ReferralCardController.ts [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+  /**
+   * Create a new payout request
+   * @param {CreateReferralCardInput} referral_card - Referral Card
+   * @returns {Promise<ReferralCard>}
+   */
+  create(referral_card: CreateReferralCardInput): Promise<ReferralCard> {
+    // what goes here?
+    return new Promise((resolve, reject) => {
+      let mutationString = `
+                      mutation ($referral_card: String! ) {
```
Comment (DenisMela) [inline src/app/controllers/ReferralCardController.ts:56]: referral_card isn't type string

File: src/app/controllers/ReferralCardController.ts [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+                          }
+                      }
+                  `;
+      this.app
+        .getAdaptor()
+        .mutate(mutationString, {
+          referral_card,
+        })
+        .then((result: MutateResult) => {
+          resolve(result.createReferralCard.id); // result does not find any Referal
```
Comment (DenisMela) [inline src/app/controllers/ReferralCardController.ts:70]: I don't understand what you mean, it's not returning a value?

File: src/app/controllers/ReferralCardController.ts [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+      let mutationString = `
+                mutation ($id: String!) {
+                  deleteReferralCard(id: $id)
+                }
+            `;
+      this.app
+        .getAdaptor()
+        .mutate(mutationString, { id })
+        .then((result: MutateResult) => {
+          resolve(result.deleteReferralCard.id); // result does not find any Referal
```
Comment (DenisMela) [inline src/app/controllers/ReferralCardController.ts:124]: This wont return an id

**Review by DenisMela — CHANGES_REQUESTED** [review]
Add this new controller to App.ts, the GraphController and the GraphQLLink

Make sure to test your mutations

File: src/app/controllers/ReferralCardController.ts (hunk 4 above)
Comment (ralph-dev) [reply src/app/controllers/ReferralCardController.ts:124]: image

@DenisMela It says it returns a string, what is the string value ??

Comment (DenisMela) [reply src/app/controllers/ReferralCardController.ts:124]: It returns `'Referral card has been successfully removed.'`

**Comment by DenisMela** [comment]
> > Add this new controller to App.ts, the GraphController and the GraphQLLink
> > Make sure to test your mutations
> 
> Can you cite the document for how to test the mutations

@rmc3408 

https://github.com/cheapreats/ts-sdk/blob/master/README.md

https://wolfbeacon.atlassian.net/wiki/spaces/CW/pages/1241514121/TS-SDK+Open+source+document

**Review by DenisMela — APPROVED** [review]
LGTM

**Pull request merged**